services/director/src/simcore_service_director/producer.py

```diff
@@ -544,7 +544,7 @@ async def _wait_for_tasks(tasks):
                 return
             await asyncio.sleep(1)  # let other events happen too
 
-    await _wait_for_tasks(tasks)
+    await _wait_for_tasks()
```

Comment:
one question here. you are deadly sure that tasks was not modified by _wait_for_tasks?

Reply:
haha! :-)
If you look above, this is an internal function. There is no need to pass an argument. Can simply use tasks in there.

Reply:
@sanderegg does this convince you? :-)

Reply:
right... it's internal.. then yeah makes total sense.